### PR TITLE
fix unwanted hour_rollover due to rounding error in circle reported pulses

### PR DESCRIPTION
**Pull request by dirixmjm**
Fix for the issue reported by @bouwew (testing required).
Rate limited approach to repeating requests on failed CircleEnergyCountersRequest

### CODE REVIEW

**Review by bouwew — DISMISSED** [review]
LGTM!

File: plugwise_usb/nodes/circle.py [hunk 1]
```diff
@@ -418,6 +428,9 @@ def _collect_energy_pulses(self, start_utc: datetime, end_utc: datetime):
                     str(self._energy_last_populated_slot),
                     str(self._energy_last_collected_timestamp),
                 )
+                if _mem_address == self._last_log_address:
+                    self._last_log_address_failed = True
+                _energy_history_failed = True
```
Comment (bouwew) [inline plugwise_usb/nodes/circle.py:433]: This line needs 1 ident less?
So that it lines up with the above `else`?

Comment (bouwew) [reply plugwise_usb/nodes/circle.py:433]: I mean line 433

Comment (dirixmjm) [reply plugwise_usb/nodes/circle.py:433]: no it is correct, 433 is executed inside the else of line 420.

Comment (bouwew) [reply plugwise_usb/nodes/circle.py:433]: See also my other feedback on Discord.

**Review by bouwew — APPROVED** [review]
LGTM!

**Pull request closed without merging**